jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/AttributesTest.java

```diff
@@ -28,13 +31,40 @@
 
 public class AttributesTest
 {
+    static Set<String> __syntheticAttributes = Set.of("Roy", "Pris", "Zhora", "Leon");
+    static AtomicReference<Object> __roy = new AtomicReference<>();
+    static AtomicReference<Object> __pris = new AtomicReference<>();
+    static AtomicReference<Object> __zhora = new AtomicReference<>();
+    static AtomicReference<Object> __leon = new AtomicReference<>();
+
     public static Stream<Attributes> attributes()
     {
         return Stream.of(
             new Attributes.Mapped(),
             new Attributes.Wrapper(new Attributes.Mapped()),
             new Attributes.Layer(new Attributes.Mapped()),
-            new AttributesMap()
+            new AttributesMap(),
+            new Attributes.Synthetic(new Attributes.Mapped())
+            {
+                @Override
+                protected Object getSyntheticAttribute(String name)
+                {
+                    return switch (name)
+                    {
+                        case "Roy" -> __roy.get();
+                        case "Pris" -> __pris.get();
+                        case "Zhora" -> __zhora.get();
+                        case "Leon" -> __leon.get();
+                        default -> null;
+                    };
+                }
+
+                @Override
+                protected Set<String> getSyntheticNameSet()
+                {
+                    return __syntheticAttributes;
+                }
+            }
```

Comment:
Seems overcomplicated.
The existing test will not test any `Synthetic` feature, and `Wrapper` is already tested, and the new test will skip all the others.
I would just make `testSynthetic()` a non-parametric test and do all the testing locally there, removing all unnecessary statics (that also are not good for parallel testing).

Reply:
I want to test that a `Synthetic` passes all the other behaviours exactly as any other.  This unit test is written as it is exactly because we have introduced different behaviours before.
I'll move the statics to be contained in a TestSynthentAttributes classs.....